pkg/proxy/proxy.go

```diff
@@ -200,8 +204,20 @@ func (s *Proxy) handleWebsocket(w http.ResponseWriter, req *http.Request) {
 		failed(err)
 		return
 	}
+	dnsNames := []string{u.Hostname()}
+	ips := []string{u.Hostname()}
+	if client.Spec.ExternalKubeAPIEnabled && len(client.Spec.Connection.ExternalKubernetesAPIEndpoint) > 0 {
+		eu, err := url.Parse(client.Spec.Connection.ExternalKubernetesAPIEndpoint)
+		if err != nil {
+			klog.Errorf("Failed to get external host %#v", err)
+		} else {
+			dnsNames = append(dnsNames, eu.Hostname())
+			ips = append(ips, eu.Hostname())
```

Comment:
I haven't tested it yet, but on AWS, default loadbalancer service assigned with an DNS name, not an IP address, is it  ok to add such DNS name to `ips`?

Reply:
this has been done by the svc watcher, if no IP exists in status field but exists a hostname filed, the `ExternalKubernetesAPIEndpoint` will use the DNS name.